scripts/giphy.py

```diff
@@ -0,0 +1,55 @@
+"""
+Respond to:
+
+    /giphy "phrase"
+
+with a random image from giphy based on that phrase.
+"""
+
+import asyncio
+import functools
+import hangups
+import logging
+import re
+import requests
+import urllib
+
+
+logger = logging.getLogger('hangbot.scripts.giphy')
+
+
+API_KEY = 'dc6zaTOxFJmzC'  # Use the public API key for now
+API_URL = 'http://api.giphy.com/v1/gifs/random'
+COMMAND = '/giphy '
+MESSAGE_REGEX = re.compile(r'^{}'.format(COMMAND), re.IGNORECASE)
```

Comment:
If you're going to require the phrase, you should either include it as a criteria for matching the command, or accept an empty phrase (like you're doing now) and log an error if the phrase is empty (instead of sending it off as an empty string (like you're doing now).

If you're going to allow for an empty phrase, then you shouldn't require the trailing space.

(And really, I should have done this too, but there's no reason to use a regex here. A simple `strip()` and `startswith` are sufficient. I'll make that change in my script as well.)